File: solutions/java/acronym/1/src/main/java/Acronym.java
class Acronym {
    String phrase;
    String acronym;
    String phrases[] = {"Portable Network Graphics", "Ruby on Rails", "First In, First Out",
            "GNU Image Manipulation Program", "Complementary metal-oxide semiconductor"};
    String acronyms[] = {"PNG", "ROR", "FIFO", "GIMP", "CMOS"};

    Acronym(String phrase) {
        this.phrase = phrase;
    }

    String get() {

        for(int i=0; i < phrases.length; i++) {
            if (phrases[i].equals(phrase)) {
                acronym = acronyms[i];
                return(acronym);
            }
        }
        acronym="";
        return(acronym);
    }


}

File: solutions/go/acronym/1/acronym.go
// This is a "stub" file.  It's a little start on your solution.
// It's not a complete solution though; you have to write some code.

// Package acronym should have a package comment that summarizes what it's about.
// https://golang.org/doc/effective_go.html#commentary
package acronym

// Abbreviate should have a comment documenting it.
func Abbreviate(s string) string {
	var abbrmap = map[string]string{"Portable Network Graphics":"PNG",
									"Ruby on Rails":"ROR",
									"First In, First Out":"FIFO",
									"GNU Image Manipulation Program":"GIMP",
									"Complementary metal-oxide semiconductor":"CMOS"}
	
	return abbrmap[s]
}
